src/fmu/tools/rms/zone_mapping.py

```diff
@@ -0,0 +1,242 @@
+from typing import Any
+
+DEBUG_OFF = 0
+DEBUG_ON = 1
+DEBUG_VERBOSE = 2
+DEBUG_VERY_VERBOSE = 3
+
+
+class ZoneMapping:
+    """
+    Keep info about available zones in modelling grid,
+    its zone numbers in the zone parameter and corresponding zone names.
+    The grid object contains info about zone names for the grid and
+    number of layers per zone as the most important info in this class.
+    The zone parameter (3D parameter with zone number per grid cell) contains
+    a table with corresponding zone name and zone number.
+    """
+
+    grid_model: Any = None
+    grid: Any = None
+    real_number: int = 0
+
+    # Key is zone index in the zones in the grid
+    grid_zone_dict: dict[int, dict] = {}
+
+    # Key is zone_number and value is zone name
+    zone_code_names: dict[int, str] = {}
+
+    # Key is zone_number
+    grid_zone_index_dict: dict[int, int] = {}
+
+    def __init__(
+        self,
+        grid_model: Any,
+        grid: Any,
+        real_number: int = 0,
+        fmu_mode: bool = False,
+        debug_level: int = DEBUG_OFF,
+    ):
+        assert grid_model
+        assert grid
+        self.grid_model = grid_model
+        self.grid = grid
+        self.real_number = real_number
+
+        if not fmu_mode:
+            # The zone parameter
+            zone_param = get_zone_parameter(
+                self.grid_model,
+                realization_number=self.real_number,
+                debug_level=debug_level,
+            )
+            self.zone_code_names = zone_param.code_names
+        else:
+            self.zone_code_names = {1: "Zone1"}
+
+        indexer = self.grid.simbox_indexer
+        zonation = indexer.zonation
+        if debug_level >= DEBUG_VERY_VERBOSE:
+            print(f"--- Zone parameter with zone_code_names: {self.zone_code_names}")
+            print(f"--- Grid with zone names: {self.grid.zone_names}")
```

Comment:
Think I have mentioned it before, but using python logging for debug messages is lightlyears better. It gives much more control and can be steered from outside while running. I can give you a brief how-to if needed.

Reply:
We could consider adressing it in own issue/PR